antarest/study/storage/rawstudy/raw_study_service.py

```diff
@@ -402,11 +402,38 @@ def unarchive(self, study: RawStudy) -> None:
         Raises:
             BadArchiveContent: If the archive is corrupted or in an unknown format.
         """
-        with open(self.get_archive_path(study), mode="rb") as fh:
+        with open(self.find_archive_path(study), mode="rb") as fh:
```

Comment:
The unarchive method does not seem to be tested anywhere:
we need to add a test that checks that we can still read both zip and 7z archives

Reply:
The `import_study` is tested and you cannot really mock it ... and most of what `unarchive` does is handled by `import_study`.

Plus there are `integration` tests for `unarchive`.

Reply:
As discussed I added a test for `import_study` with an archived study in `.7z` format